kinoml/features/complexes.py

```diff
@@ -0,0 +1,544 @@
+            A list of OpenEye molecules for writing.
+        path: str
+            File path for saving molecules.
+        """
+        from openeye import oechem
+
+        with oechem.oemolostream(path) as ofs:
+            for molecule in molecules:
+                oechem.OEWriteMolecule(ofs, molecule)
+        return
+
+    @staticmethod
+    def _has_ligand(molecule: oechem.OEGraphMol) -> bool:
+        """
+        Check if OpenEye molecule contains a ligand.
+        Parameters
+        ----------
+        molecule: oechem.OEGraphMol
+            An OpenEye molecule.
+        Returns
+        -------
+        bool
+            True if molecule has ligand, False otherwise.
+        """
+        from openeye import oechem
+
+        ligand = oechem.OEGraphMol()
+        protein = oechem.OEGraphMol()
+        water = oechem.OEGraphMol()
+        other = oechem.OEGraphMol()
+        oechem.OESplitMolComplex(ligand, protein, water, other, molecule)
+
+        if ligand.NumAtoms() > 0:
+            return True
+
+        return False
+
+    @staticmethod
+    def _prepare_complex(
+        protein_ligand_complex: oechem.OEGraphMol,
+        electron_density: Union[oegrid.OESkewGrid, None] = None,
+        loop_db: Union[str, None] = None,
+    ) -> List[Union[oechem.OEGraphMol, None]]:
+        """
+        Prepare an OpenEye molecule holding a protein ligand complex for docking.
+        Parameters
+        ----------
+        protein_ligand_complex: oechem.OEGraphMol
+            An OpenEye molecule holding a structure with protein and ligand.
+        electron_density: oegrid.OESkewGrid
+            An OpenEye grid holding the electron density.
+        loop_db: str or None
+            Path to OpenEye Spruce loop database.
+        Returns
+        -------
+        protein: oechem.OEGraphMol
+            An OpenEye molecule holding a prepared protein structure.
+        ligand: oechem.OEGraphMol
+            An OpenEye molecule holding a prepared ligand structure.
+        """
+        from openeye import oechem, oespruce
+
+        # create design units
+        structure_metadata = oespruce.OEStructureMetadata()
+        design_unit_options = oespruce.OEMakeDesignUnitOptions()
+        if loop_db is not None:
+            design_unit_options.GetPrepOptions().GetBuildOptions().GetLoopBuilderOptions().SetLoopDBFilename(
+                loop_db
+            )
+        if electron_density is None:
+            design_units = list(
+                oespruce.OEMakeDesignUnits(
+                    protein_ligand_complex, structure_metadata, design_unit_options
+                )
+            )
+        else:
+            design_units = list(
+                oespruce.OEMakeDesignUnits(
+                    protein_ligand_complex,
+                    electron_density,
+                    structure_metadata,
+                    design_unit_options,
+                )
+            )
+        if len(design_units) == 1:
```

Comment:
Can this be combined to `if len(design_units) >=1:` ?

Reply:
true, but I think this differentiation might be important for logging or metadata, although not implemented yet